### ISSUE TITLE
Something is not precise in function binary_cross_entropy in tensorlayer.cost package

### ISSUE THREAD

**Issue by DallasBuyer** [zsdonghao/tensorlayer#129]
In binary_cross_entropy function, now that "loss(x, z) = - sum_i (x[i] * log(z[i]) + (1 - x[i]) * log(1 - z[i]))", I think the code should be written like the following form:
return tf.reduce_mean(tf.reduce_sum(-(target * tf.log(output + epsilon) + (1. - target) * tf.log(1. - output + epsilon)), axis=1).
But the original form is as follows:
return tf.reduce_mean(-(target * tf.log(output + epsilon) + (1. - target) * tf.log(1. - output + epsilon))).
Of course, it makes no difference when target and output are one dimension,.
Actually, I'm not very sure weather it is right about what i say above.

**Comment by zsdonghao** [comment]
Thank you for your report. 
Could you make a push request, this update is your contribution.

**Comment by DallasBuyer** [comment]
OK
